sql/src/test/java/io/crate/analyze/SelectStatementAnalyzerTest.java

```diff
@@ -612,11 +611,25 @@ public void testSelectDistinctWithGroupBy() {
     }
 
     @Test
+    public void testSelectDistinctWithGroupByLimitAndOffset() {
+        SelectAnalyzedStatement analysis =
+            analyze("select distinct max(id) from users group by name order by 1 limit 5 offset 10");
+        assertThat(analysis.relation(), instanceOf(QueriedSelectRelation.class));
+        assertThat(analysis.relation().querySpec(),
+                   isSQL("SELECT doc.users.max(id) GROUP BY doc.users.max(id) " +
+                         "ORDER BY doc.users.max(id) LIMIT 5 OFFSET 10"));
+        QueriedSelectRelation outerRelation = (QueriedSelectRelation) analysis.relation();
+        assertThat(outerRelation.subRelation(), instanceOf(QueriedDocTable.class));
+        assertThat(outerRelation.subRelation().querySpec(),
+                   isSQL("SELECT max(doc.users.id) GROUP BY doc.users.name ORDER BY max(doc.users.id)"));
```

Comment:
I was thinking that to keep the order by also on the subRelation is somehow beneficial, but I was wrong...